examples/action_dann_lightn/model.py

```diff
@@ -60,49 +64,202 @@ def get_feat_extractor(model_name, num_classes, num_channels):
-    elif model_name == "R3D_18":
-        feature_network = r3d_18(pretrained=True)
-        feature_dim = 512
-    elif model_name == "R2PLUS1D_18":
-        feature_network = r2plus1d_18(pretrained=True)
-        feature_dim = 512
-    elif model_name == "MC3_18":
-        feature_network = mc3_18(pretrained=True)
-        feature_dim = 512
+    if attention in attention_list:
+        att = True
+    elif attention == "None":
+        att = False
     else:
-        raise ValueError("Unsupported model: {}".format(model_name))
-    return feature_network, feature_dim
+        raise ValueError("Wrong attention. Current: {}".format(attention))
+
+    if image_modality == "rgb":
+        if model_name == "I3D":
+            pretrained_model = "rgb_imagenet"
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_i3d_joint(
+                    rgb_pt=pretrained_model, flow_pt=None, num_classes=num_classes, attention=attention, pretrained=True
+                )
+            else:
+                logging.info("No SELayer.")
+                feature_network = i3d_joint(
+                    rgb_pt=pretrained_model, flow_pt=None, num_classes=num_classes, pretrained=True
+                )
+            # model.replace_logits(num_classes)
+            class_feature_dim = 1024
+            dmn_feature_dim = class_feature_dim
+
+        elif model_name == "R3D_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_r3d(rgb=True, flow=False, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = r3d(rgb=True, flow=False, pretrained=True)
+            class_feature_dim = 512
+            dmn_feature_dim = class_feature_dim
+
+        elif model_name == "R2PLUS1D_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_r2plus1d(rgb=True, flow=False, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = r2plus1d(rgb=True, flow=False, pretrained=True)
+            class_feature_dim = 512
+            dmn_feature_dim = class_feature_dim
+
+        elif model_name == "MC3_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_mc3(rgb=True, flow=False, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = mc3(rgb=True, flow=False, pretrained=True)
+            class_feature_dim = 512
+            dmn_feature_dim = class_feature_dim
+
+        else:
+            raise ValueError("Unsupported model: {}".format(model_name))
+
+    elif image_modality == "flow":
+        if model_name == "I3D":
```

Comment:
The same switches again. Consider to put modality more inner. See `loaddata.digit_access` etc how to do these elegantly.